config/drop-legacy-python.py

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env', 'python3
+from shared.call import call
+from shared.call import wait_for_accept
+from shared.git import get_branch_name
+from shared.git import git_branch
+from shared.path import change_dir
+import argparse
+import collections
+import os
+import pathlib
+import shutil
+import sys
+import toml
+
+
+parser = argparse.ArgumentParser(
+    description='Drop support of Python 2.7 up to 3.6 from a package.')
+parser.add_argument(
+    'path', type=pathlib.Path, help='path to the repository to be configured')
+parser.add_argument(
+    '--branch',
+    dest='branch_name',
+    default=None,
+    help='Define a git branch name to be used for the changes. If not given'
+         ' it is constructed automatically and includes the configuration'
+         ' type')
+
+
+args = parser.parse_args()
+path = args.path.absolute()
+
+if not (path / '.git').exists():
+    raise ValueError('`path` does not point to a git clone of a repository!')
+if not (path / '.meta.toml').exists():
+    raise ValueError('The repository `path` points to has no .meta.toml!')
+
+with change_dir(path) as cwd_str:
+    cwd = pathlib.Path(cwd_str)
+    bin_dir = cwd / 'bin'
+    meta_cfg = collections.defaultdict(dict, **toml.load('.meta.toml'))
+    config_type = meta_cfg['meta']['template']
+    branch_name = get_branch_name(args.branch_name, config_type)
+    updating = git_branch(branch_name)
+
+    call(bin_dir / 'bumpversion', '--breaking', '--no-input')
+    call(bin_dir / 'addchangelogentry',
+         'Drop support for Python 2.7, 3.5, 3.6.', '--no-input')
+    call(bin_dir / 'check-python-versions',
+         '--drop=2.7,3.5,3.6', '--only=setup.py')
+    print('Remove legacy Python specific settings from .meta.toml')
+    call(os.environ['EDITOR'], '.meta.toml')
+
+    config_package_args = [
+        sys.executable,
+        'config-package.py',
+        path,
+        f'--branch={branch_name}',
+        '--no-push',
+    ]
+    call(*config_package_args, cwd=cwd_str)
+    print('Remove `six` from the list of dependencies and other Py 2 things.')
+    call(os.environ['EDITOR'], 'setup.py')
+    src = path.resolve() / 'src'
+    call('find', src, '-name', '"*.py"',
+         '-exec', 'bin/pyupgrade', '--py3-plus', '--py37-plus', '{}', ';')
```

Comment:
For me on Mac, this does not do anything. But I get no error either, so it is hard to say what is wrong. I guess it is a difference between Mac and Linux.

When I try it myself on the command line I do get an error:

```
$ find /Users/maurits/community/plone-coredev/6.0/src/Products.CMFCore/src -name "*.py" -exec bin/pyupgrade --py3-plus --py37-plus {} ;
find: -exec: no terminating ";" or "+"
```

I tried more quotes, but that did not help. I can never remember or figure out how to use find and exec this way...

I have a `pyupgrade.sh` script that does the following, and that works for me, and does lots of changes after your call has run:

```
git ls-files | grep '\.py$' | xargs pyupgrade $*
```

Reply:
> When I try it myself on the command line I do get an error:

Shell eats the `;` so you need to quote it:

    $ find /Users/maurits/community/plone-coredev/6.0/src/Products.CMFCore/src -name "*.py" -exec bin/pyupgrade --py3-plus --py37-plus {} ";"

> I tried more quotes, but that did not help.

Where did you put those quotes?

> I can never remember or figure out how to use find and exec this way...

I like

    find ... -exec command {} +

because it runs the command once, passing all the filenames as arguments, and you don't need to quote the `+`.

I've no idea if pyupgrade supports multiple filenames on the command line.

Reply:
On the command line the `;` has to be quoted. I normally use `\;` so the shell does not interpret it as end of the command. About using `+` I just learnt today.

I am also on a Mac. Probably the `'"*.py"'` has to become `*.py` as there is no shell to interpret the quotes.

Reply:
I tried quotes at least in these ways I think:

    "pyupgrade --py3-plus --py37-plus" {} ;
    "pyupgrade --py3-plus --py37-plus {}" ;
    "pyupgrade --py3-plus --py37-plus {} ;"

Michael is probably right that the double quotes around `'"*.py"'` were not needed. The latest code on this branch works.

`+` can be good, but every now and then I try to use either `find -exec` or `xargs` in situations where there are too many files, so for example even a simple `ls *` actually fails, say in a buildout downloads cache or shared eggs cache. With `+`, `find` would fail as well, as it would try to fit all arguments on one line. So `";"` is good then.

Reply:
Good idea. @mauritsvanrees